INT-3753: Add Channel Send Count to IMBE

garyrussell opened the pull request:
JIRA: https://jira.spring.io/browse/INT-3753

Missing accessor for channel send counts.

**cherry-pick to 4.1.x**

garyrussell committed:
INT-3753: Expose More Metrics via IMBE

JIRA: https://jira.spring.io/browse/INT-3753

INT-3753: Add Channel Send Count to IMBE

Missing accessor for channel send counts.

Expose Full Metrics Objects

artembilan commented on spring-integration-jmx/src/main/java/org/springframework/integration/monitor/IntegrationMBeanExporter.java:
```diff
@@ -630,6 +630,18 @@ public long getSourceMessageCountLong(String name) {
+
+	public long getChannelSendCountLong(String name) {
+		if (channelsByName.containsKey(name)) {
+			return channelsByName.get(name).getSendCountLong();
```
So, `getSendErrorCount()` will we cover on demand?
Wouldn't it be better just expose `getChannelMetrics(String name)` ?

garyrussell replied:
I will add the error count and also expose metrics objects.

garyrussell commented:
Rebased, squashed, pushed.

When cherry-picking, fix minor conflicts in test case and remove

    		assertSame(intermediate, messageChannelsMonitor.getChannelMetrics(intermediateChannelName));

artembilan commented:
Merging...

artembilan commented:
... and cherry-picked with mentioned conflict fix
